### ISSUE TITLE
hasData don't return true just after calling setData

### ISSUE THREAD

**Issue by mcsky** [biig-io/ngx-smart-modal#129]
```
[ x ] bug report 
[ ] feature request
[ ] support request
```

**Current behavior**
When you set data to your modal and call `hasData` at the next line you'll get a `false`.
It isn't a really common case but may lead to confusion.

**Expected behavior**

When some data is set on the modal the function `hasData` should return `true`

**Reproduction of the problem**

```javascript
const modal = modalService.getModal('myModal');
modal.hasData() // return false, it's ok
modal.setData('foo') 
modal.hasData() // Return false, wuutttt
setTimeout(() => console.log(modal.hasData()), 0); // Print yes
```

* **Browser:** [all  ]

**Comment by mcsky** [comment]
https://github.com/biig-io/ngx-smart-modal/blob/5fee67eac3b0543e1ee7fbfe6f574d41ef1c654d/src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts#L198-L205

Why there is a setTimeout here @maximelafarie? :)

**Comment by maximelafarie** [comment]
@mcsky this is for triggering a new Angular cycle. I'll check if I can improve it! 😉

**Comment by maximelafarie** [comment]
Closed by #143
